src/fides/api/service/connectors/sql_connector.py

```diff
@@ -184,15 +185,26 @@ def mask_data(
         query_config = self.query_config(node)
         update_ct = 0
         client = self.client()
+
+        # Check if safe_mode is enabled
+        db: Session = Session.object_session(self.configuration)
+        config_proxy = ConfigProxy(db)
+        safe_mode_enabled = getattr(config_proxy.execution, "safe_mode", False)
+
         for row in rows:
             update_stmt: Optional[TextClause] = query_config.generate_update_stmt(
                 row, policy, privacy_request
             )
             if update_stmt is not None:
-                with client.connect() as connection:
-                    self.set_schema(connection)
-                    results: LegacyCursorResult = connection.execute(update_stmt)
-                    update_ct = update_ct + results.rowcount
+                if safe_mode_enabled:
+                    # In safe mode, log the SQL statement instead of executing it
+                    logger.warning(f"SAFE MODE - Would execute SQL: {update_stmt}")
```

Comment:
👏 for (starting to) add this more generically than just BQ... it's a good pattern for us.

there are other derived connectors though that override mask_data themselves in an unsafe way, though.